src/dist/betaprime.rs

```diff
@@ -0,0 +1,653 @@
+    use rand_xoshiro::Xoshiro256Plus;
+
+    use super::*;
+    use crate::test_basic_impls;
+
+    const TOL: f64 = 1E-12;
+
+    test_basic_impls!(f64, BetaPrime, BetaPrime::new(1.0, 1.0).unwrap());
+
+    #[test]
+    fn new() {
+        let bp = BetaPrime::new(1.2, 3.4).unwrap();
+        assert::close(bp.alpha, 1.2, TOL);
+        assert::close(bp.beta, 3.4, TOL);
+
+        assert!(BetaPrime::new(0.0, 1.0).is_err());
+        assert!(BetaPrime::new(-1.0, 1.0).is_err());
+        assert!(BetaPrime::new(1.0, 0.0).is_err());
+        assert!(BetaPrime::new(1.0, -1.0).is_err());
+        assert!(BetaPrime::new(f64::INFINITY, 1.0).is_err());
+        assert!(BetaPrime::new(1.0, f64::INFINITY).is_err());
+        assert!(BetaPrime::new(f64::NAN, 1.0).is_err());
+        assert!(BetaPrime::new(1.0, f64::NAN).is_err());
+    }
+
+    #[test]
+    fn mean_when_beta_gt_one() {
+        let mut rng = Xoshiro256Plus::seed_from_u64(123);
+        let bp = BetaPrime::new(2.0, 3.0).unwrap();
+
+        // Theoretical mean
+        let mu_theory: f64 = bp.mean().unwrap();
+
+        // Sample mean
+        let n = 1000;
+        let mut s = 0.0;
+        for _ in 0..n {
+            let x: f64 = bp.draw(&mut rng);
+            s += x;
+        }
+        let mu_sample = s / n as f64;
+
+        assert::close(mu_sample, mu_theory, 0.03);
+    }
+
+    #[test]
+    fn mean_when_beta_leq_one() {
+        let bp = BetaPrime::new(2.0, 1.0).unwrap();
+        assert!(bp.mean().is_none());
+    }
+
+    #[test]
+    fn variance_when_beta_gt_two() {
+        let mut rng = Xoshiro256Plus::seed_from_u64(123);
+        let bp = BetaPrime::new(10.0, 15.0).unwrap();
+
+        // Theoretical variance
+        let var_theory: f64 = bp.variance().unwrap();
+
+        // Sample variance assuming correct mean
+        let mean = bp.mean().unwrap();
+        let n = 1000;
+        let mut sse = 0.0;
+        for _ in 0..n {
+            let x: f64 = bp.draw(&mut rng);
+            sse += (x - mean).powi(2);
+        }
+
+        // Calculate variance
+        let var_sample: f64 = sse / n as f64;
+
+        assert::close(var_sample, var_theory, 0.01);
+    }
+
+    #[test]
+    fn variance_when_beta_leq_two() {
+        let bp = BetaPrime::new(2.0, 2.0).unwrap();
+        assert!(bp.variance().is_none());
+    }
+
+    #[test]
+    fn mode_when_alpha_geq_one() {
+        let bp = BetaPrime::new(2.0, 3.0).unwrap();
+        assert::close(bp.mode().unwrap(), 0.25, TOL);
+    }
+
+    #[test]
+    fn mode_when_alpha_lt_one() {
+        let bp = BetaPrime::new(0.5, 3.0).unwrap();
+        assert!(bp.mode().unwrap() == 0.0);
+    }
+
+    #[test]
+    fn draw_should_return_positive_values() {
+        let mut rng = rand::thread_rng();
+        let bp = BetaPrime::new(2.0, 3.0).unwrap();
+        for _ in 0..100 {
+            let x: f64 = bp.draw(&mut rng);
+            assert!(x > 0.0);
+        }
+    }
+
+    // TODO: Uncomment once limiting behavior is corrected
```

Comment:
Should we mark this as ignored with a message?

Reply:
Oh like unimplemented! ?

Reply:
More something like this: https://doc.rust-lang.org/reference/attributes/testing.html#the-ignore-attribute

Reply:
@schmidmt I dropped the entropy code for now. Don't know that we'll use it anyway, and it's easy enough to reimplement later if we need it